apps/aecore/src/aec_blocks.erl

```diff
@@ -142,7 +143,24 @@ to_header(#block{height = Height,
 serialize_for_network(B = #block{}) ->
     {ok, jsx:encode(serialize_to_map(B))}.
 
+serialize_client_readable(B) ->
+    serialize_to_map(B, [txs_as_objects]).
+
 serialize_to_map(B = #block{}) ->
+    serialize_to_map(B, []).
+
+serialize_to_map(B = #block{}, Options) ->
+    SerializeTxFun =
+        case lists:member(txs_as_objects, Options) of
```

Comment:
May I suggest rather having the fun passed directly as argument.
My main concern (apart from readability) is potential weakening of the Dialyzer type checks.

Reply:
I see amended - thanks.